From 99b1ab90b0dfd9905d5496af1d82afb71ee9b856 Mon Sep 17 00:00:00 2001
From: Michael Butler <butlermichael@google.com>
Date: Tue, 18 May 2021 23:04:36 -0700
Subject: [PATCH 1/3] Remove libneuralnetworks_shim.so

The NNAPI team is delaying our updatable platform driver plans to
Android T. This CL removes libneuralnetworks_shim.so, as it is no longer
needed for Android S.

Bug: 188587379
Test: mma
Change-Id: I64216a61063a3dfe3b4d4c21ef45f72e5fd5aef6
---
 apex/Android.bp                            |  1 -
 shim_and_sl/Android.bp                     | 46 ----------------------
 shim_and_sl/libneuralnetworks_shim.map.txt | 32 ---------------
 3 files changed, 79 deletions(-)
 delete mode 100644 shim_and_sl/libneuralnetworks_shim.map.txt

diff --git a/apex/Android.bp b/apex/Android.bp
index eab5f7a26..879555d38 100644
--- a/apex/Android.bp
+++ b/apex/Android.bp
@@ -41,7 +41,6 @@ apex_defaults {
     androidManifest: ":com.android.neuralnetworks-androidManifest",
     native_shared_libs: [
         "libneuralnetworks",
-        "libneuralnetworks_shim",
     ],
     compile_multilib: "both",
     key: "com.android.neuralnetworks.key",
diff --git a/shim_and_sl/Android.bp b/shim_and_sl/Android.bp
index dbb28cf57..041c91ecb 100644
--- a/shim_and_sl/Android.bp
+++ b/shim_and_sl/Android.bp
@@ -91,52 +91,6 @@ cc_library_static {
     ],
 }
 
-cc_library_shared {
-    name: "libneuralnetworks_shim",
-    llndk_stubs: "libneuralnetworks_shim.llndk",
-    version_script: "libneuralnetworks_shim.map.txt",
-    apex_available: [
-        "com.android.neuralnetworks",
-        "test_com.android.neuralnetworks",
-    ],
-    licenses: ["Android-Apache-2.0"],
-    vendor_available: true,
-    min_sdk_version: "30",
-    whole_static_libs: [
-        "libneuralnetworks_shim_static",
-    ],
-    header_libs: [
-        "libneuralnetworks_headers",
-    ],
-    static_libs: [
-        "android.hardware.common-V2-ndk_platform",
-        "android.hardware.neuralnetworks-V1-ndk_platform",
-        "libcutils",
-        "neuralnetworks_supportlibrary_loader",
-        "neuralnetworks_types",
-        "neuralnetworks_utils_hal_aidl",
-        "neuralnetworks_utils_hal_common",
-    ],
-    shared_libs: [
-        "libbase",
-        "libbinder_ndk",
-        "libhidlbase",
-        "libhidlmemory",
-        "liblog",
-        "libnativewindow",
-    ],
-    export_include_dirs: [
-        "public",
-    ],
-}
-
-llndk_library {
-    name: "libneuralnetworks_shim.llndk",
-    symbol_file: "libneuralnetworks_shim.map.txt",
-    export_include_dirs: ["public"],
-    licenses: ["Android-Apache-2.0"],
-}
-
 cc_library_static {
     name: "neuralnetworks_supportlibrary_loader",
     srcs: [
diff --git a/shim_and_sl/libneuralnetworks_shim.map.txt b/shim_and_sl/libneuralnetworks_shim.map.txt
deleted file mode 100644
index 159d663be..000000000
--- a/shim_and_sl/libneuralnetworks_shim.map.txt
+++ /dev/null
@@ -1,32 +0,0 @@
-#
-# Copyright (C) 2021 The Android Open Source Project
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-#  Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-
-# This lists the entry points visible to applications that use the
-# Neural Networks lib. Other entry points present in the library won't be
-# usable.
-
-LIBNEURALNETWORKS_SHIM {
-  global:
-    ANeuralNetworksShim_registerSupportLibraryService; # llndk
-    ANeuralNetworksShimDeviceInfo_create; # llndk
-    ANeuralNetworksShimDeviceInfo_free; # llndk
-    ANeuralNetworksShimDeviceInfo_setCapabilities; # llndk
-    ANeuralNetworksShimDeviceInfo_setNumberOfCacheFilesNeeded; # llndk
-    ANeuralNetworksShimDeviceInfo_addVendorExtension; # llndk
-
-  local:
-    *;
-};
-- 
2.17.1

